filament/backend/include/backend/platforms/PlatformMetal.h

```diff
@@ -39,6 +39,22 @@ class PlatformMetal final : public Platform {
     Driver* createDriver(void* sharedContext, const Platform::DriverConfig& driverConfig) noexcept override;
     int getOSVersion() const noexcept override { return 0; }
 
+    /**
+     * Optionally initializes the Metal platform by acquiring resources necessary for rendering.
+     *
+     * This method attempts to acquire a Metal device and command queue, returning true if both are
+     * successfully obtained, or false otherwise. Typically, these objects are acquired when
+     * the Metal backend is initialized. This method allows clients to check for their availability
+     * earlier.
+     *
+     * Calling initialize() is optional and safe to do so multiple times. After initialize() returns
+     * true, subsequent calls will continue to return true but have no effect.
+     *
+     * @returns true if the device and command queue have been successfully obtained; false
+     * otherwise.
+     */
+    bool initialize() noexcept;
```

Comment:
which thread do you expect the user to call this from? 

For vk, the platform methods are meant to be called from the backend thread.  But this seems like a client-facing API?

Reply:
yes, and on top of that the method doesn't look thread safe, so we should definitely specify the threading model for this method in the doc.

Reply:
Yeah, great points, definitely missed this.